openedx_learning/apps/core/itemstore/models.py

```diff
@@ -0,0 +1,133 @@
+"""
+The item model hiearachy is: Item -> ItemVersion -> ItemInfo -> ItemRaw
+
+Item is the versionless thing that is guaranteed to exist for the lifetime of
+the LearningContext. An ItemVersion is a different version of that item for a
+given LearningContext, and may include policy changes (like grading). ItemInfo
+covers basic metadata that is intrinsic to the item itself, and now how it's
+used in a LearningContext. ItemRaw represents the raw byte data.
+
+TODO: Add link to ADR after it merges.
```

Comment:
Ack, sorry, I am struggling to rebuild this hierarchy in my head. I'm going to try to rephrase it here. Let me know if I'm off:
* ...Assume "LC" is some LearningContext, and "LCV" some LearningContextVersion within LC.
* An ItemRaw is a blob of byte data and a mimetype.
  * Example: an ItemRaw instance "PR", containing some OLX file
* An ItemInfo associates an ItemRaw with metadata, including its LearningContext and its item type.
  * Example: an ItemInfo instance "PI", which tells us that PR is a `problem` and belongs to LC.
* An Item is an atomic block of content associted with a LC. In isolation, an Item does not tell us about its content, because content is associated with versions and an Item is versionless.
  * Example: an Item instance "P" in LC.
 * An ItemVersion joins together an abstract Item instance with a concrete ItemInfo, denoting a particular version of the content, policy, and metadata of an Item.
   * Example: an ItemVersion instance "PV", joining together P and PI.
   * Example: an ItemVersion instance "PV2", joining together P and PI, with some updated policy.
   * Example: a later ItemVersion instance "PVx", joining together P and some modified ItemInfo "PIx".
   * and so on
 * LearningContextVersionItemVersions relate LearningContextVersions to ItemVersions. This table is the connective tissue telling us which ItemVersions are in a LearningContextVersion (and, by corollary, which Items are in a LearningContext).
   * Example: PV is in LCV
   * Example: PV2 is in "LCV2", a later version of LC
   * Example: PV2 is also in "LCV3", a later version of LC that updates some Item(s) other than P